### PR TITLE
Fix 279 - Adding Import Command

### LINKED ISSUE #279: Proposal: `import` command
`reckoner import <release_name> --chart_repository  <chart_repository> --namespace <namespace>` would retrieve use `helm get values` and `helm list` to retrieve information about a release in a cluster.  It would output yaml to stdout that could be used in a `course.yml` to assist in Reckoner being used in clusters with current Helm releases.

### PR DESCRIPTION
**Pull request by ejether**
This adds an import command

Fixes #279

This is a first draft/sketch of the feature for feedback purposes. If this look acceptable, I'll add some tests and docs.

### CODE REVIEW

**Comment by sudermanjr** [comment]
I love the concept. There's a couple issues with it right now. 

1) I would love it if it was pre-indented on the output. That way `>>` would add it to a course.yml. A bit of a nit.
2) There seems to be a bug

```
✗ reckoner import --release_name gke-node-termination-handler --namespace kube-system --repository fairwinds-stable
2021-02-02 20:35:46  root[37217]  WARNING  Import is experimental and may be unreliable. Double check all output.
2021-02-02 20:35:46  root[37217]     INFO  Found Helm Version 3.3.4
2021-02-02 20:35:49  root[37217]     INFO  Found Helm Version 3.3.4
gke-node-termination-handler:
  chart: gke
  repository: fairwinds-stable
  version: node
  namespace: kube-system
  ```
  
  ```
  ✗ helm ls -n kube-system
NAME                            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                                   APP VERSION
gke-node-termination-handler    kube-system     5               2021-02-02 20:16:33.197338 -0700 MST    deployed        gke-node-termination-handler-1.2.1      1.0
```

**Comment by ejether** [comment]
This does not support charts with more than one dash in them. Please rename your chart. 
s/

How does the code structure and interaction look? It feels `dumb` but sometimes people call that elegant?

**Comment by sudermanjr** [comment]
> How does the code structure and interaction look? It feels `dumb` but sometimes people call that elegant?

I would say this is fine for MVP. It feels a little fragile to be outputting the release yaml that way (if we change the release schema in any way, it won't match anymore). But for now, this works for me.

Always curious what @lucasreed thinks these days too :-)

**Comment by lucasreed** [comment]
Cool! Seems good for an MVP, lets get those docs and tests 😄

**Comment by ejether** [comment]
Aye Aye

**Comment by ejether** [comment]
READY!

File: README.md [hunk 1]
```diff
@@ -57,6 +57,10 @@ reckoner plot course.yaml --run-all
 
 Grafana and Polaris should now be installed on your cluster!
 
+## Adopting Reckoner
+If you're already using Helm but want to start using `reckoner`, you can use `reckoner import` to facilitate your migration!
```
Comment (sudermanjr) [inline README.md:61]:
```suggestion
Warning: Experimental

If you're already using Helm but want to start using `reckoner`, you can use `reckoner import` to facilitate your migration.

We recommend carefully examining the output of a `reckoner diff` before relying on any imported course.yml definitions.
```

File: reckoner/cli.py [hunk 2]
```diff
@@ -346,6 +351,23 @@ def lint(ctx, log_level, course_file=None):
         ctx.exit(1)
     click.echo(click.style("No schema validation errors found.", fg="green"))
 
+
+@cli.command(name='import')  # import is special word in python, need to rename command here
+@click.pass_context
+@log_level_option
+@click.option("--release_name", help='The release name to import', required=True)
+@click.option("--namespace", help='The namespace of the release', required=True)
```
Comment (sudermanjr) [inline reckoner/cli.py:359]:
```suggestion
@click.option("--namespace", help='The namespace of the release being imported', required=True)
```

File: reckoner/cli.py [hunk 3]
```diff
@@ -346,6 +351,23 @@ def lint(ctx, log_level, course_file=None):
...
+@click.option("--repository", help='The repository of the chart', required=True)
```
Comment (sudermanjr) [inline reckoner/cli.py:360]:
```suggestion
@click.option("--repository", help='The repository of the chart being imported', required=True)
```

File: reckoner/cli.py [hunk 4]
```diff
@@ -346,6 +351,23 @@ def lint(ctx, log_level, course_file=None):
...
+def import_release(ctx, log_level, release_name, namespace, repository):
+    """Outputs a chart block that can be used to import the specified release"""
```
Comment (sudermanjr) [inline reckoner/cli.py:362]:
```suggestion
    """Outputs a release block that can be used to import the specified release"""
```

File: reckoner/importer.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
+# -- coding: utf-8 --
```
Comment (sudermanjr) [inline reckoner/importer.py:1]: ?

File: reckoner/importer.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    for _release in json.loads(response.stdout):
+        if _release.get('name') == release:
+            return _release
+
+    raise ReckonerException(f"Release {release} not found in {namespace}")
+
+
+def draft_release(release: str, namespace: str, repository: str) -> dict:
+    """
+    Parses release information and release values and parses it together into a dictionary
```
Comment (sudermanjr) [inline reckoner/importer.py:74]:
```suggestion
    Parses release information and values, then parses them together into a dictionary
```

File: reckoner/importer.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    with the specified release, namespace, and repository
```
Comment (sudermanjr) [inline reckoner/importer.py:75]:
```suggestion
    with the specified release, namespace, repository, and values
```

File: reckoner/tests/test_importer.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+from reckoner.importer import get_values, list_release
+import unittest
+from unittest import mock
+
+# from reckoner.helm.client import HelmClientException
+# from reckoner.reckoner import Reckoner, ReckonerInstallResults
+# from reckoner.chart import ChartResult
+# from reckoner.exception import ReckonerCommandException, NoChartsToInstall, ReckonerException
+# from reckoner.helm.client import get_helm_client
+
```
Comment (sudermanjr) [inline reckoner/tests/test_importer.py:24]:
```suggestion
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]
This is super awesome. I think I ended up only recommending changes to documentation and such, to clarify a few things. Otherwise I still love the functionality, and the testing is great too. :-D

File: reckoner/importer.py (hunk 5 above)
Comment (ejether) [reply reckoner/importer.py:1]: I copied it from elsewhere in the repo. 
It's peppy
https://www.python.org/dev/peps/pep-0263/

**Review by sudermanjr — APPROVED** [review]

**Pull request merged**